src/components/popups/pause-popup/pause-popup-component.ts

```diff
@@ -11,21 +14,51 @@ export const PAUSE_POPUP_EVENT_DATA = {
 export class PausePopupComponent extends BasePopupComponent {
   selectLevelButton?: BaseButtonComponent;
   restartLevelButton?: BaseButtonComponent;
-
+  confirmPopup?: ConfirmPopupComponent;
+  protected audioPlayer = new AudioPlayer();
+  protected clickTimeout?;
   protected override id = 'pause-popup';
 
   onInit() {
     const targetId = this.contentContainerId;
     // TODO integrate buttons
-    this.selectLevelButton = new MapButton({ targetId });
+    this.selectLevelButton = new MapButton({ id: 'map-button', className: 'me-4', targetId });
     this.selectLevelButton.onClick(() => this.handleClick('select-level'));
     this.restartLevelButton = new RetryButtonHtml({ targetId });
     this.restartLevelButton.onClick(() => this.handleClick('restart-level'));
+    this.confirmPopup = new ConfirmPopupComponent({
+      hideClose: true
+    });
+    
+    this.audioPlayer = new AudioPlayer();
+  }
+
+  handleClick(data: any) {
+    if (lang === 'english') {
+      const unsub = this.confirmPopup.onClose((confirmData) => {
+        unsub();
+        this.handleConfirmClose(confirmData, data);
+      });
+
+      if (this.clickTimeout) clearTimeout(this.clickTimeout);
```

Comment:
Can we consider Clear Timeout and Set timeout in a seperate method for reusability and readability?

Reply:
added `lodash-es` using `debuonce` function for this instead, for reusability.